Applications/Utils/MeshGeoTools/ComputeSurfaceNodeIDsInPolygonalRegion.cpp

```diff
@@ -0,0 +1,161 @@
+/**
+ * @file ComputeSurfaceNodeIDsInPolygonalRegion.cpp
+ * @brief Computes mesh node ids of mesh nodes within a polygonal region, that resides on the surface.
+ *
+ * @copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "tclap/CmdLine.h"
+#include "logog/include/logog.hpp"
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/LogogSimpleFormatter.h"
+#include "BaseLib/StringTools.h"
+#include "BaseLib/FileTools.h"
+
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/readGeometryFromFile.h"
+
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/Polygon.h"
+
+#include "MathLib/Vector3.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/MeshSurfaceExtraction.h"
+
+void writeToFile(std::string const& id_area_fname, std::string const& csv_fname,
+	std::vector<std::pair<std::size_t, double>> const& ids_and_areas,
+	std::vector<MeshLib::Node*> const& mesh_nodes)
+{
+	std::ofstream ids_and_area_out(id_area_fname);
+	if (!ids_and_area_out) {
+		ERR("Unable to open the file \"%s\" - aborting.", id_area_fname.c_str());
+		std::abort();
+	}
+	std::ofstream csv_out(csv_fname);
+	if (!csv_out) {
+		ERR("Unable to open the file \"%s\" - aborting.", csv_fname.c_str());
+		std::abort();
+	}
+
+	ids_and_area_out << std::setprecision(20);
+	csv_out << std::setprecision(20);
+
+	ids_and_area_out << ids_and_areas[0].first << " " << ids_and_areas[0].second;
```

Comment:
- Fixed twice output.
- Added size check for `ids_and_areas` outside in the main function.